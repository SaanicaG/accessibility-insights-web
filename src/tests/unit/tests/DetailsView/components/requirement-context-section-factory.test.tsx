// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import { AutomatedChecks } from 'assessments/automated-checks/assessment';
import { RequirementContextSectionDeps } from 'DetailsView/components/requirement-context-section';

import {
    RequirementContextSectionFactoryProps,
    getRequirementContextSectionForAssessment,
    getRequirementContextSectionForQuickAssess,
} from 'DetailsView/components/requirement-view-context-section-factory';
import { ContentPage, ContentPageComponent } from 'views/content/content-page';
import { ContentPanelButton } from 'views/content/content-panel-button';
import { mockReactComponents } from '../../../mock-helpers/mock-module-helpers';

jest.mock('views/content/content-panel-button');
describe('RequirementContextSectionFactoryTest', () => {
    let props: RequirementContextSectionFactoryProps;
    beforeEach(() => {
        mockReactComponents([ContentPanelButton]);
        props = {
            deps: {} as RequirementContextSectionDeps,
            className: 'test-class-name',
            assessmentKey: 'test-assessment-key',
            whyItMatters: ContentPage.create(() => 'WHY IT MATTERS' as any),
            infoAndExamples: { pageTitle: 'test-page-title' } as ContentPageComponent,
            helpfulResourceLinks: [{ href: 'test-resource-href', text: 'test-resource-text' }],
        };
    });

    describe('getRequirementContextSectionForAssessment', () => {
        it('returns null', () => {
            expect(getRequirementContextSectionForAssessment(props)).toBeNull();
        });
    });

    describe('getRequirementContextSectionForQuickAssess', () => {
        it('renders content from props', () => {
            const renderResult = render(getRequirementContextSectionForQuickAssess(props));
            expect(renderResult.asFragment()).toMatchSnapshot();
        });

        it('returns null if we are in Automated Checks', () => {
            props.assessmentKey = AutomatedChecks.key;
            expect(getRequirementContextSectionForQuickAssess(props)).toBeNull();
        });
    });
});
