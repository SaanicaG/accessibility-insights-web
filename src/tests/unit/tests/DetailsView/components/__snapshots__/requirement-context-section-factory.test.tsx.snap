// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequirementContextSectionFactoryTest getRequirementContextSectionForQuickAssess renders content from props 1`] = `
<DocumentFragment>
  <div
    class="test-class-name"
  >
    <section
      class="requirementContextContainer"
    >
      <h3
        class="contextHeading"
      >
        Why it matters
      </h3>
      WHY IT MATTERS
      <h3
        class="contextHeading"
      >
        Helpful resources
      </h3>
      <div
        class="contextLinks"
      >
        <mock-contentpanelbutton
          contenttitle="Info and examples"
          deps="[object Object]"
          reference="[object Object]"
        >
          Info and examples
        </mock-contentpanelbutton>
        <div
          class="helpLink"
        >
          <div
            class="ms-TooltipHost insightsTooltipHost root-109"
            role="none"
          >
            <a
              class="ms-Link insights-link insightsLink root-110"
              href="test-resource-href"
              target="_blank"
            >
              test-resource-text
            </a>
            <div
              hidden=""
              id="tooltip0"
              style="position: absolute; width: 1px; height: 1px; margin: -1px; padding: 0px; border: 0px; overflow: hidden; white-space: nowrap;"
            />
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
